src/libs/bin_decoder/lv_bin_decoder.c

```diff
@@ -150,171 +153,25 @@ lv_result_t lv_bin_decoder_info(lv_image_decoder_t * decoder, const void * src,
-        lv_image_dsc_t * image = (lv_image_dsc_t *)dsc->src;
-        if(image->data == NULL) {
-            return LV_RESULT_INVALID;
-        }
-
-        lv_color_format_t cf = image->header.cf;
-        if(dsc->header.flags & LV_IMAGE_FLAGS_COMPRESSED) {
-            res = decode_compressed(decoder, dsc);
-        }
-        else if(LV_COLOR_FORMAT_IS_INDEXED(cf)) {
-            /*Need decoder data to store converted image*/
-            decoder_data_t * decoder_data = get_decoder_data(dsc);
-            if(decoder_data == NULL) {
-                return LV_RESULT_INVALID;
-            }
-
-            res = decode_indexed(decoder, dsc);
-        }
-        else if(LV_COLOR_FORMAT_IS_ALPHA_ONLY(cf)) {
-            /*Alpha only image will need decoder data to store pointer to decoded image, to free it when decoder closes*/
-            decoder_data_t * decoder_data = get_decoder_data(dsc);
-            if(decoder_data == NULL) {
-                return LV_RESULT_INVALID;
-            }
-
-            res = decode_alpha_only(decoder, dsc);
-        }
-        else {
-            /*In case of uncompressed formats the image stored in the ROM/RAM.
-             *So simply give its pointer*/
-
-            decoder_data_t * decoder_data = get_decoder_data(dsc);
-            lv_draw_buf_t * decoded = &decoder_data->c_array;
-            dsc->decoded = decoded;
-            lv_draw_buf_from_image(decoded, image);
-
-            if(decoded->header.stride == 0) {
-                /*Use the auto calculated value from decoder_info callback*/
-                decoded->header.stride = dsc->header.stride;
-            }
-
-            /**
-             * @todo need to convert c-array image stride if not match
-             *
-             * lv_draw_buf_create(); //create new draw buf that meets requirement
-             * lv_draw_buf_copy(); //copy from c-array image to new draw buf
-             */
-            uint32_t stride_expect = lv_draw_buf_width_to_stride(dsc->header.w, dsc->header.cf);
-            if(dsc->header.stride != stride_expect) {
-                LV_LOG_WARN("Stride mismatch");
-#if 0
-                /**
-                 * @fixme ignore for now
-                 */
-                free_decoder_data(dsc);
-                return LV_RESULT_INVALID;
-#endif
-            }
-
-            res = LV_RESULT_OK;
-        }
-    }
-
-    if(res != LV_RESULT_OK) {
-        free_decoder_data(dsc);
-        return res;
-    }
-
+    bool create_res = bin_decoder_decode_data(dsc);
+    if(create_res == false) return LV_RESULT_INVALID;
     if(dsc->decoded == NULL) return LV_RESULT_OK; /*Need to read via get_area_cb*/
 
     /*Add it to cache*/
-    t = lv_tick_elaps(t);
-    lv_cache_lock();
-    lv_cache_entry_t * cache = lv_cache_add(NULL, 0, decoder->cache_data_type, dsc->header.w * dsc->header.h * 4);
-    if(cache == NULL) {
-        lv_cache_unlock();
+    lv_image_decoder_cache_data_t search_key;
```

Comment:
It won't cost much since it only compare if the first character is printable or not.

Use `src` only will make user application much convenient to invalidate cache:
```c
/* From user's application code */
/*Image changed, invalidate cache now*/
lv_image_cache_invalide(src);
```

Reply:
ok fine